### PR TITLE
Don't show hovercards for primitive and null values

### PR DESCRIPTION
**Pull request by elliette**
Fixes https://github.com/flutter/devtools/issues/4871

Example: 

For the following, hovering on the variable name will show a hovercard with the value. However, hovering on the value itself ("Hello world!", 10, 2.33) will not show a hovercard.

```
final myString = "Hello world!";
final myInt = 10;
final myDouble = 2.33;
```

### CODE REVIEW

**Review by bkonyi — APPROVED** [review]

File: packages/devtools_app/lib/src/screens/debugger/codeview.dart [hunk 1]
```diff
@@ -1172,7 +1172,7 @@ class _LineItemState extends State<LineItem>
 
     final word = wordForHover(event.localPosition.dx, widget.lineContents);
 
-    if (word != '') {
+    if (word != '' && !isPrimitiveValueOrNull(word)) {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/debugger/codeview.dart:1175]: can this be `word.isNotEmpty` instead?

File: packages/devtools_app/lib/src/shared/ui/hover.dart [hunk 2]
```diff
@@ -66,6 +66,20 @@ String wordForHover(double dx, TextSpan line) {
   return word;
 }
 
+bool isPrimitiveValueOrNull(String valueAsString) {
+  final isNull = valueAsString == 'null';
+  final isBool = valueAsString == 'true' || valueAsString == 'false';
+  final isInt = int.tryParse(valueAsString) != null;
+  final isDouble = double.tryParse(valueAsString) != null;
+  final firstChar = valueAsString[0];
+  final lastChar = valueAsString[valueAsString.length - 1];
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/ui/hover.dart:75]: can we guarantee the string is non-empty at this point? should we add an assert?

**Review by kenzieschmoll — APPROVED** [review]

File: packages/devtools_app/lib/src/shared/ui/hover.dart (hunk 2 above)
Comment (elliette) [reply packages/devtools_app/lib/src/shared/ui/hover.dart]: verified that it's an appropriate length to do a string check

File: packages/devtools_app/lib/src/screens/debugger/codeview.dart (hunk 1 above)
Comment (elliette) [reply packages/devtools_app/lib/src/screens/debugger/codeview.dart:1175]: will follow up with word is not empty check here

**Pull request merged**